cranelift/interpreter/src/step.rs

```diff
@@ -118,6 +119,31 @@ where
         })
     };
 
+    let branch_indexed = |index: V| -> ControlFlow<V> {
+        let (table, default_target) = match inst {
+            InstructionData::BranchTable {
+                table, destination, ..
+            } => (table, Some(destination)),
+            InstructionData::IndirectJump { table, .. } => (table, None),
+            _ => unreachable!(),
+        };
+        let jt_data = &state.get_current_function().jump_tables[table];
+
+        let index = index.into_int().unwrap();
+
+        // Convert to usize to remove negative indexes from the following operations
+        let jump_target = usize::try_from(index)
+            .ok()
+            .and_then(|i| jt_data.as_slice().get(i))
+            .copied()
+            .or(default_target);
+
+        match jump_target {
+            Some(block) => ControlFlow::ContinueAt(block, SmallVec::new()),
+            None => ControlFlow::Trap(CraneliftTrap::User(TrapCode::IndirectCallToNull)),
```

Comment:
I just noticed that this is probably not the most applicable trap code (call instead of jump?), should I create a new one?